scripts/contractInteraction/staking_vesting.py

```diff
@@ -515,6 +540,26 @@ def addVestingCodeHash(vestingLogic):
     data = staking.addContractCodeHash.encode_input(vestingLogic)
     sendWithMultisig(conf.contracts['multisig'], staking.address, data, conf.acct)
 
+# addresses is vesting addresses that will be set
+# vestingCreationAndTypes is the array of VestingCreationAndTypeDetail struct
+# struct VestingCreationAndTypeDetail {
+#         bool isSet;
+#         uint32 vestingType;
+#         uint128 vestingCreationType;
+#     }
+# make sure the length of addresses & vestingCreationAndTypeDetails is the same
+def registerVestingToVestingCreationAndTypes(addresses, vestingCreationAndTypeDetails):
+    if len(addresses) != len(vestingCreationAndTypeDetails):
+        raise Exception("umatched length of array between addresses & vestingCreationAndTypeDetails")
+
+    vestingRegistry = Contract.from_abi("VestingRegistryLogic", address=conf.contracts['VestingRegistryProxy'], abi=VestingRegistryLogic.abi, owner=conf.acct)
+    data = vestingRegistry.registerVestingToVestingCreationAndTypes.encode_input(addresses, vestingCreationAndTypeDetails)
+    sendWithMultisig(conf.contracts['multisig'], vestingRegistry.address, data, conf.acct)
+
+def getVestingCreationAndTypes(vestingAddress):
+    vestingRegistry = Contract.from_abi("VestingRegistryLogic", address=conf.contracts['VestingRegistryProxy'], abi=VestingRegistryLogic.abi, owner=conf.acct)
+    print(vestingRegistry.vestingCreationAndTypes(vestingAddress))
+
 def readTokenOwner(vestingAddress):
-    vesting = Contract.from_abi("VestingLogic", address=vestingAddress, abi=VestingLogic.abi, owner=conf.acct)
-    print(vesting.tokenOwner())
\ No newline at end of file
+    vesting = Contract.from_abi("VestingLogic", address=vestingAddress, abi=VestingRegistryLogic.abi, owner=conf.acct)
```

Comment:
while it will work, the ABI used in confusing 
pls change for VestingLogic?

Reply:
updated